Update debian ISO to v11.7.0

alexpeterson91 opened the pull request:
Update debian ISO to v11.7.0 to test with NUC12's to see if it resolves the firmware issue currently resolved only by workaround in the experiemental ISO release

alexpeterson91 committed:
Update debian ISO to v11.7.0

Update debian ISO to v11.7.0 to test with NUC12's to see if it resolves the firmware issue currently resolved only by workaround in the experiemental ISO release

pablomendezroyo commented:
@alexpeterson91 nice catch!

- Has been tested the ISO installation with this new Debian ISO?
- Does the new debian image fixes the issue with wifi drivers? in any case we must keeping updating the Debian ISO, so this shouldnt be a blocker for merging this PR


cc: @3alpha @dsimog01

dsimog01 commented:
> 

I already talked to @alexpeterson91 and he told me the drivers issue is still not fixed by this Debian version. However, I agree we should test and merge this anyway

3alpha commented on iso/scripts/generate_dappnode_iso_debian.sh:
```diff
@@ -1,12 +1,13 @@
 ISO_PATH="/images/${ISO_NAME}"
-ISO_URL=https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/archive/11.5.0+nonfree/amd64/iso-cd/
-SHASUM="ce1dcd1fa272976ddc387554202013e69ecf1b02b38fba4f8c35c8b12b8f521e  ${ISO_PATH}"
+ISO_URL=https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/current/amd64/iso-cd/
```
```suggestion
ISO_URL=https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/11.7.0+nonfree/amd64/iso-cd/
```

We shouldn't use current because current changes. :)

alexpeterson91 replied:
thank you i was writing this out in my last comment but then noticed your collapsed commit and edited my post since i asked if there was a permalink already.

Commit:
Update iso/scripts/generate_dappnode_iso_debian.sh

alexpeterson91 committed:
update from test 2

alexpeterson91 commented:
> > 
> 
> I already talked to @alexpeterson91 and he told me the drivers issue is still not fixed by this Debian version. However, I agree we should test and merge this anyway

yes totally agreed, If 11.7 doesn't have any breaking changes to dependencies of core DNPs etc we should certainly upgrade to it ASAP
Details for everyone's reference regarding  Debian 11.7:
As I relayed via Diego already, unfortunately it does not solve all the driver issues that came up with NUC12 and 13s,+, but it does solve a LOT of them, including an edge case that only a few current users (obviously I was the first to notice it by being the first Nodler to use a NUC with x16 PCI-e Slot like end of 2021) even would need to worry about, the issue being that the NUCs with the slots randomly throw errors on startup and boot and then just on the console over days about the PCI-e drivers not working among other things  i collected a bunch of these errors, including HD audio, wifi, some storage drivers and the 2 Ethernet cards as issues that needed to be fixed for the NUC12 experimental firmware.  

Diego's branch resolved all the issues we found except for the PCI-e issues, but has backports and a bunch of firmware files manually added.

Testing Debian 11.6 resolved none of the issues at all.

Testing Debian 11.7 here **Resolved**:

1. Ethernet Card Firmware for Intel 2.5Gbe Port 
2. Ethernet Card Firmware for Marvell 10Gbe Port 
3. PCI-E Firmware (have not tested with a expansion card yet but errors are all gone and don't recur. Until now we hadn't even found a workaround for it, though Diego and I put little to no effort into this particular component)

Debian 11.7 did **NOT** Resolve:

1. Intel Wifi 6E AX211 NVMe Card drivers, not detected by debian without manually installing driver, This is the only remaining blocker to resolve NUC12+ issues though since wired internet works with this new Debian 11.7 distro, adding a script installing just the Wifi driver for this particular card is trivial, I think...
2. Intel/Realtek HD-Audio Sound Card Driver/firmware/Codec (not an issue to worry about really, nothing currently uses audio in any way on Dappnode AFAIK) issue is just the error which pops up on the console of NUC models with the HD sound card, but even that can be suppressed by disabling the sound card in the BIOS. 
_Also we do have the missing wifi and sound card drivers already in Diego's Branch And the biggest issue for NUC12's was no way to get internet, with 11.7 we get wired internet on NUC12s so we can much more easily get NUC12's working with Wi-Fi using 11.7_

Finally,  regarding this link to the Debian netisnt.ISO  In this PR, I've had to edit  it and change the URL path to the permalink for 11.5 when 11.6 was released, so I reverted this to the "current" URL folder just because i was testing it for NUC12s not thinking bigger picture at the time i opened this PR.  So as soon as Debian 11.8 or anything supersedes Debian 11.7 (like Debian 11.6 when it was released and it broke all the CI and CD stuff since the ISO was pointing to the current folder not the archive permalink for Debian 11.7 and kept failing . I don't know if there is already an archive for it in addition to the current on the debian mirrors, but if so i will change it to a permalink so it stays as is and doesnt break when a new version is out.  
Or separately as @pablomendezroyo we need to keep upgrading Debian versions, we should add a bump Debian iso upstream version workflow to check stable releases, on a cron schedule like weekly or something to check the current URL or hash signature and see if it resolves or matches, and if not open a PR with the new filename and signature. and of course run all the build checks on that PR as well.

alexpeterson91 committed:
revert Update docker-compose.yml

revert personal changes 

This is/was a personal test branch and i have been testing it offline and saved my local work on my branch and pushed it and then when went back to github i found that @dappnode/core had started to move on it themselves so my last commit from my VS-Code wasnt made as a PR because I didnt know anyone was planning on implmenting it just yet for everyone especially

alexpeterson91 commented on .dappnode_profile:
```diff
@@ -1,9 +1,9 @@
+export IPFS_VERSION="${IPFS_VERSION:-0.2.19}"
 export VPN_VERSION="${VPN_VERSION:-0.2.8}"
-export DAPPMANAGER_VERSION="${DAPPMANAGER_VERSION:-0.2.58}"
+export DAPPMANAGER_VERSION="${DAPPMANAGER_VERSION:-0.2.66}"
```
revert personal changes 

This is/was a personal test branch and i have been testing it offline and saved my local work on my branch and pushed it and then when went back to github i found that @dappnode/core had started to move on it themselves so my last commit from my VS-Code wasnt made as a PR because I didnt know anyone was planning on implementing it just yet for everyone especially, but more importantly regarding the profile here, why is this not being updated.  I know for the pre-release theres a workflow to enter all values, but thats not for normal users, normal users follow the dev build instructions with `docker-compose build && docker-compose up` which uses these values, when the NUC12 issues first started appearing as people bought them, they kept trying to do this  and kept having to update so many core versions some major making it a process to get up to date, from their own docker generated ISOs. 
I also know things have been changing in the way a lot is organized, being built, automated QA (which is awesome,  ive found several issues so far so i wanna learn more on how its configured and i can hopefully copy all workflows for dappnode so i can replicate them in my organization without using your runners, time, and secrets etc. and dont bloat up your branches with my own tests.) but shouldn't we be keeping core DNP versions up to date across the board? not just in DNP_CORE but everywhere they're listed and used such as here?

alexpeterson91 commented on docker-compose.yml:
```diff
@@ -7,7 +7,7 @@ services:
       - BUILD=false # In case you want to re-generate a all the images, not recommended
       - CLEAN=true # it remove the images directory
-      - UNATTENDED=false # UNATTENDED version
+      - UNATTENDED=true # UNATTENDED version
```
revert personal changes 

again
This is/was a personal test branch and i have been testing it offline and saved my local work on my branch and pushed it and then when went back to github i found that @dappnode/core had started to move on it themselves so my last commit from my VS-Code wasnt made as a PR because I didnt know anyone was planning on implementing it just yet for everyone especially.
Reverting this one now

alexpeterson91 commented:
Debian 12 supercedes this in PR #575

pablomendezroyo commented:
If this PR is not needed, then please, remove it in favor of the Debian 12 one
